### PR TITLE
Improve interface of generated code

### PR DESCRIPTION
**Pull request by joelfisch**
- Do not catch exceptions for user
- Improve monad containing configuration
- Make the monad operation function the default

See https://github.com/Haskell-OpenAPI-Code-Generator/Haskell-OpenAPI-Client-Code-Generator/issues/9

The changes made by this PR are best visible in the changed Stripe example: https://github.com/Haskell-OpenAPI-Code-Generator/Stripe-Haskell-Library/pull/3/files#diff-7f6666bd3cb88c8284fab86dfb0de42d

In addition, this is a screenshot of the Haddock for `GetEvents`:

image

### CODE REVIEW

**Comment by NorfairKing** [comment]
This looks so much better, I'm excited!!

File: .circleci/testing/level2/petstore-running-example/test/Spec.hs [hunk 1]
```diff
@@ -57,8 +57,8 @@ main =
       it "runUpdatePet" $ do
         let requestExpectation = expectBody "{\"photoUrls\":[],\"status\":null,\"category\":null,\"name\":\"Harro\",\"id\":null,\"tags\":null}" $ expectMethod "PUT" noExpectation
         response <- runMock runUpdatePet (requestExpectation, succeededResponse)
-        (getResponseBody . fromRight undefined) response `shouldBe` UpdatePetResponse200
+        getResponseBody response `shouldBe` UpdatePetResponse200
       it "runUpdatePetWithTag" $ do
         let requestExpectation = expectBody "{\"name\":\"Tag 1\",\"id\":3}" $ expectMethod "PUT" noExpectation
         response <- runMock runUpdatePetWithTag (requestExpectation, succeededResponse)
-        (getResponseBody . fromRight undefined) response `shouldBe` UpdatePetResponse200
+        getResponseBody response `shouldBe` UpdatePetResponse200
```
Comment (NorfairKing) [inline .circleci/testing/level2/petstore-running-example/test/Spec.hs:64]: Isn't this so much nicer :D

Comment (joelfisch) [reply .circleci/testing/level2/petstore-running-example/test/Spec.hs:64]: Yes, definitely 😉

**Comment by joelfisch** [comment]
IMO, the most important issues which introduce big breaking changes are done. Now I was wondering: Are there more critical issues which you would like to have resolved before updating the Stripe library and adapting your code to check if everything works fine for you?

**Comment by NorfairKing** [comment]
@joelfisch We should probably try it out together in pair-programming on a call. Then we'll discover the issues together. No need to update the hackage library until we get that done. I can just depend on a repo.

**Comment by joelfisch** [comment]
Perfect, this sounds good 👍 

Are you available this evening?

**Comment by NorfairKing** [comment]
@joelfisch Yes, feel free to schedule something. See cs-syd.eu/agenda.

**Pull request merged**